src/Configuration/Config.Azure.KeyVault.Secrets/src/AzureKeyVaultConfigurationProvider.cs

```diff
@@ -0,0 +1,171 @@
+    {
+        private readonly TimeSpan? _reloadInterval;
+        private readonly SecretClient _client;
+        private readonly IKeyVaultSecretManager _manager;
+        private Dictionary<string, LoadedSecret> _loadedSecrets;
+        private Task _pollingTask;
+        private readonly CancellationTokenSource _cancellationToken;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="AzureKeyVaultConfigurationProvider"/>.
+        /// </summary>
+        /// <param name="client">The <see cref="SecretClient"/> to use for retrieving values.</param>
+        /// <param name="manager">The <see cref="IKeyVaultSecretManager"/> to use in managing values.</param>
+        /// <param name="reloadInterval">The timespan to wait in between each attempt at polling the Azure KeyVault for changes. Default is null which indicates no reloading.</param>
+        public AzureKeyVaultConfigurationProvider(SecretClient client, IKeyVaultSecretManager manager, TimeSpan? reloadInterval = null)
+        {
+            _client = client ?? throw new ArgumentNullException(nameof(client));
+            _manager = manager ?? throw new ArgumentNullException(nameof(manager));
+            if (reloadInterval != null && reloadInterval.Value <= TimeSpan.Zero)
+            {
+                throw new ArgumentOutOfRangeException(nameof(reloadInterval), reloadInterval, nameof(reloadInterval) + " must be positive.");
+            }
+
+            _pollingTask = null;
+            _cancellationToken = new CancellationTokenSource();
+            _reloadInterval = reloadInterval;
+        }
+
+        /// <summary>
+        /// Load secrets into this provider.
+        /// </summary>
+        public override void Load() => LoadAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+
+        private async Task PollForSecretChangesAsync()
+        {
+            while (!_cancellationToken.IsCancellationRequested)
+            {
+                await WaitForReload();
+                try
+                {
+                    await LoadAsync();
+                }
+                catch (Exception)
+                {
+                    // Ignore
+                }
+            }
+        }
+
+        protected virtual Task WaitForReload()
+        {
+            // WaitForReload is only called when the _reloadInterval has a value.
+            return Task.Delay(_reloadInterval.Value, _cancellationToken.Token);
+        }
+
+        private async Task LoadAsync()
+        {
+            var secretPages = _client.GetPropertiesOfSecretsAsync();
+
+            var tasks = new List<Task<Response<KeyVaultSecret>>>();
+            var newLoadedSecrets = new Dictionary<string, LoadedSecret>();
+            var oldLoadedSecrets = Interlocked.Exchange(ref _loadedSecrets, null);
+
+            await foreach (var secret in secretPages.ConfigureAwait(false))
+            {
+                if (!_manager.Load(secret) || secret.Enabled != true)
```

Comment:
Probably should check `Enabled` first. Avoid problems down the road, and the fastest check in this expression.